### DISCUSSION TITLE
How to print parsing error?

### DISCUSSION THREAD

**Discussion by shilangyu** [pest-parser/pest#713] [Q&A] [answered]
Readme mentions ["meaningful error reporting"](https://github.com/pest-parser/pest#meaningful-error-reporting) however I have not found how to print such a pretty-printed error. Looked through docs.rs, the book, and examples.

**Comment by tomtau** [comment, marked as answer]
The default `Display` should work:

```rust
    let result = YourParser::parse(Rule::your_rule, &input);
    println!("{}", result.unwrap_err());
```

**Reply by shilangyu** [reply]
Ah great. The most obvious solution. Thanks!
